tools/NEC/nec_sd_generator.py

```diff
@@ -0,0 +1,1128 @@
+    element_phases += list(reversed(element_phases))
+    print("Element phases:\n{}".format(element_phases))
+    return element_phases
+
+
+def calculate_circular_phase(frequency_hz, antenna_spacing_m, num_antennas):
+    """
+    Calculate circular phase for antenna array
+    :param frequency_hz: What is the frequency used in Hz?
+    :param antenna_spacing_m: What is the straight line antenna spacing in meters?
+    :param num_antennas: How many antennas to calculate phases for?
+    :return: The list of phase delays for each antenna
+    """
+    # TODO: Complete this method
+    raise NotImplementedError("Circular phase not implemented")
+
+
+def frequency_card(num_freq_steps=1, start_frequency=8.0, freq_step_size=0.5):
+    """
+    :param num_freq_steps: How many frequency steps to calculate output for, default 1
+    :param start_frequency: What is the starting frequency in MHz? Default 8
+    :param freq_step_size: What is the frequency step size in MHz? Default 0.5
+    :return: NEC compatible frequency calculation card
+    """
+    return "FR 0 {} 0 0 {} {}\r\n".format(num_freq_steps, start_frequency, freq_step_size)
+
+
+def radiation_pattern_card(theta_start=45.0, theta_increment=0.0, theta_steps=1,
+                           phi_start=0.0, phi_increment=1.0, phi_steps=361):
+    """
+    :param theta_start: Start elevation angle in degrees. Default 45.0
+    :param theta_increment: Elevation increment angle in degrees. Default 0.0
+    :param theta_steps: How many steps in elevation to calculate. Default 1
+    :param phi_start: Start azimuth angle in degrees. Default 0.0
+    :param phi_increment: Azimuth increment angle in degrees. Default 1.0
+    :param phi_steps: How many steps in azimuth to calculate. Default 361
+    :return: NEC compatible radiation pattern calculation card.
+    """
+    return "RP 0 {} {} 1000 {} {} {} {}\r\n".format(theta_steps, phi_steps, theta_start,
+                                                    phi_start, theta_increment, phi_increment)
+
+
+def ground_card(ground_type=2, epse=13, conductivity=0.005):
+    """
+    :param ground_type: See NEC user manual. 2 is Sommerfeld ground. 0 finite gnd, 1 perfect gnd
+    :param epse: See NEC user manual. 13 is relative dielectric constant for gnd in vicinity
+    :param conductivity: Conductivity in mhos/meter in the case of perfect ground
+    :return: The ground card NEC compatible string
+    """
+    return "GN {} 0 0 0 {} {}\r\n".format(ground_type, epse, conductivity)
+
+
+def extended_kernel_card():
+    """
+    :return: The extended kernel card NEC compatible string
+    """
+    return "EK\r\n"
+
+
+def end_of_run_card():
+    """
+    :return: The end of run card NEC compatible string
+    """
+    return "EN"
+
+
+def max_coupling_card():
+    """
+    :return: The max coupling card NEC compatible string
+    """
+    return "CP * * * *\r\n"
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
```

Comment:
you can also specify your own usage message. If you create a separate function for it, you can do this: 
`    parser = argparse.ArgumentParser(usage=usage_msg())`

This might be nice for describing what all this script does (inputs and what it outputs, as well as what is implemented and not implemented at this time). 

if you place the usage_msg() function at the top of the file it also will help those reading the code to see what it does.